packages/tc-schema-validator/tests/types.js

```diff
@@ -1,386 +1,53 @@
-							).toBe(true);
-						});
-						it('has valid type', () => {
-							expect(config.type).toBeDefined();
-							expect(config.type).toMatch(
-								arrayToRegExp(validPropTypes),
-							);
-						});
-
-						it('has valid fieldset', () => {
-							if (config.fieldset) {
-								expect(validFieldsetNames).toContain(
-									config.fieldset,
-								);
-							}
-						});
-
-						it('has valid deprecation reason', () => {
-							if (config.deprecationReason) {
-								expect(typeof config.deprecationReason).toBe(
-									'string',
-								);
-								expect(config.deprecationReason).not.toMatch(
-									/\n/,
-								);
-							}
-						});
-
-						if (!typeNames.includes(config.type)) {
-							describe('direct property', () => {
-								// tests for direct properties
-								it('may be required', () => {
-									if (config.required) {
-										expect(config.required).toBe(true);
-									}
-								});
-								it('may be an identifier', () => {
-									if (config.canIdentify) {
-										expect(config.canIdentify).toBe(true);
-									}
-								});
-								it('may define a pattern', () => {
-									if (config.pattern) {
-										expect(config.pattern).toMatch(
-											validStringPatternsRX,
-										);
-									}
-								});
-
-								it('may hide inactive records', () => {
-									if (config.showInactive) {
-										expect(typeof config.showInactive).toBe(
-											'boolean',
-										);
-									}
-								});
-
-								it('may allow writing of inactive records', () => {
-									if (config.writeInactive) {
-										expect(
-											typeof config.writeInactive,
-										).toBe('boolean');
-									}
-								});
-
-								it('may define true and false labels', () => {
-									if (config.trueLabel || config.falseLabel) {
-										expect(typeof config.trueLabel).toBe(
-											'string',
-										);
-										expect(typeof config.falseLabel).toBe(
-											'string',
-										);
-										expect(config.type).toBe('Boolean');
-									}
-								});
-
-								it('may define examples', () => {
-									if (config.examples) {
-										expect(
-											Array.isArray(config.examples),
-										).toBe(true);
-									}
-								});
-							});
-						} else {
-							const RELATIONSHIP_NAME = getStringValidator(
-								'RELATIONSHIP_NAME',
-							);
-							describe('relationship property', () => {
-								it('must specify underlying relationship', () => {
-									expect(config.relationship).toMatch(
-										RELATIONSHIP_NAME,
-									);
-								});
-								it('must specify direction', () => {
-									expect(config.direction).toMatch(
-										/^incoming|outgoing$/,
-									);
-								});
-								it('may be hidden', () => {
-									if (config.hidden) {
-										expect(config.hidden).toBe(true);
-									}
-								});
+describe('validate types', () => {
+	types.filter(({ from, to }) => !from && !to).forEach(typeTestSuite);
```

Comment:
I don't understand why these filters are still here now that types and relationshipTypes are split into different getters?

Reply:
You are right. I should access each via `sdk.rawData.getTypes()` and `sdk.rawData.getRelationshipTypes()` without filtering